src/flow.ts

```diff
@@ -390,18 +392,30 @@ export class Flow {
         else parentFunction.tempExternrefs = temps = [];
         break;
       }
-      case <u32>NativeType.Exnref: {
-        let tempExnrefs = parentFunction.tempExnrefs;
-        if (tempExnrefs) temps = tempExnrefs;
-        else parentFunction.tempExnrefs = temps = [];
-        break;
-      }
       case <u32>NativeType.Anyref: {
         let tempAnyrefs = parentFunction.tempAnyrefs;
         if (tempAnyrefs) temps = tempAnyrefs;
         else parentFunction.tempAnyrefs = temps = [];
         break;
       }
+      case <u32>NativeType.Eqref: {
+        let tempEqrefs = parentFunction.tempEqrefs;
+        if (tempEqrefs) temps = tempEqrefs;
+        else parentFunction.tempEqrefs = temps = [];
+        break;
+      }
+      case <u32>NativeType.I31ref: {
+        let tempI31refs = parentFunction.tempI31refs;
+        if (tempI31refs) temps = tempI31refs;
+        else parentFunction.tempI31refs = temps = [];
+        break;
+      }
+      case <u32>NativeType.Dataref: {
```

Comment:
btw I'm wondering why we need case to `u32` for switch all cases?

Reply:
Yes, because `NativeType` is `usize` (can be a pointer to a tuple for example), but all the canonical ones fit into 32 bits.